src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -343,6 +345,29 @@ private void finishCurrentDeployment() throws InterruptedException {
                         }
                     }
                     deploymentDirectoryManager.persistLastSuccessfulDeployment();
+                } else if (DeploymentStatus.REJECTED.equals(deploymentStatus)) {
+                    if (result.getFailureCause() != null) {
+                        updateStatusDetailsFromException(statusDetails, result.getFailureCause(),
+                                currentDeploymentTaskMetadata.getDeploymentType());
+                        logger.atWarn().setCause(result.getFailureCause()).kv(DEPLOYMENT_ID_LOG_KEY_NAME, deploymentId)
+                                .kv(GG_DEPLOYMENT_ID_LOG_KEY_NAME, ggDeploymentId)
+                                .kv(DEPLOYMENT_DETAILED_STATUS_KEY, result.getDeploymentStatus())
+                                .kv(DEPLOYMENT_ERROR_STACK_KEY, statusDetails.get(DEPLOYMENT_ERROR_STACK_KEY))
+                                .kv(DEPLOYMENT_ERROR_TYPES_KEY, statusDetails.get(DEPLOYMENT_ERROR_TYPES_KEY))
+                                .log("Deployment task rejected with following errors");
+                    }
+
+                    deploymentStatusKeeper.persistAndPublishDeploymentStatus(deploymentId, ggDeploymentId,
+                            configurationArn, type, JobStatus.REJECTED.toString(), statusDetails, rootPackages);
+
+                    if (currentDeploymentTaskMetadata.getDeploymentTask() instanceof KernelUpdateDeploymentTask) {
+                        try {
+                            kernel.getContext().get(KernelAlternatives.class).rollbackCompletes();
+                        } catch (IOException e) {
+                            logger.atError().log("Failed to reset Nucleus rollback directory", e);
+                        }
+                    }
```

Comment:
why are we doing this? We never ran the deployment because we rejected it, so why are we dealing with rollbacks?

Reply:
Makes sense. I will remove it!